lms/resources/_js_config/file_picker_config.py

```diff
@@ -11,8 +11,10 @@ class FilePickerConfig:
     @classmethod
     def d2l_config(cls, request, _application_instance):
         """Get D2L files config."""
-        files_enabled = request.product.settings.files_enabled
-
+        files_enabled = (
```

Comment:
Yes, files are enabled for "this" LMS but "this" might not be D2L right now.

We could send the title for the button here ("D2L files" | "Blackboard files") or the frontend could do that itself.

For now this is a straightforward fix.

Reply:
I think this comment caused a bit of confusion, so I'll try and explain. 

Our setting structure is a two level deep naming scheme where we have an outer and inner name like `canvas.sections_enabled`, `canvas.files`, or `microsoft_onedrive.files_enabled`.

I think the issue we are having is that in the `<a>.<b>` scheme the type of thing that `<a>` is changes. In some cases it's a document source (`microsoft_onedrive`) in others it's a product, and in an increasing number it's both at the same time. This confusion is extended into the product system which uses these keys now. This last part is the confusion I would be keen to eliminate.

This is just an example to show the idea, not an actual suggestion to fix it, but if the concepts were differentiated we might have:

* `PRODUCT:canvas.sections_enabled`
* `DOCUMENT_SOURCE:canvas_files.enabled`
* `DOCUMENT_SOURCE:microsoft_onedrive.enabled`

Of course there is a link between the document source "Canvas files" and the product "Canvas" we would still want and need to represent and use, but this would make a distinction between the two.

This also naturally leaves space for a second type of file integration for a given product: Canvas for example has different ways of storing files of which we only integrate with one.